### PR TITLE
1856: Include a link to the webrev comment in the pull request's description

### PR DESCRIPTION
**Pull request by zhaosongzs**
Some users complained about that GitHub often hide the webrev comment and it's hard for them to find webrev comment among hundreds of comments. 

In this patch, the pr bot would add the link to webrev comment to the pr body.

Originally, the comments from mlbridge bot would not trigger the update of the pr, now, when mlbridge bot **first** post the webrev comment, it will trigger the update of the pr.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace

### Issue
 * [SKARA-1856](https://bugs.openjdk.org/browse/SKARA-1856): Include a link to the webrev comment in the pull request's description

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**) ⚠️ Review applies to [0746297f](https://git.openjdk.org/skara/pull/1494/files/0746297fc9489b72dc3ccbfae14ac9c497a7401a)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/skara.git pull/1494/head:pull/1494` \
`$ git checkout pull/1494`

Update a local copy of the PR: \
`$ git checkout pull/1494` \
`$ git pull https://git.openjdk.org/skara.git pull/1494/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1494`

View PR using the GUI difftool: \
`$ git pr show -t 1494`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/skara/pull/1494.diff">https://git.openjdk.org/skara/pull/1494.diff</a>

</details>

### CODE REVIEW

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckWorkItem.java [hunk 1]
```diff
@@ -94,6 +96,12 @@ String getMetadata(CensusInstance censusInstance, String title, String body, Lis
                                         .flatMap(comment -> comment.body().lines())
                                         .filter(line -> METADATA_COMMENTS_PATTERN.matcher(line).find())
                                         .collect(Collectors.joining());
+            commentString = commentString + comments.stream()
+                    .filter(comment -> comment.author().username().equals("mlbridge[bot]"))
```
Comment (zhaosongzs) [inline bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckWorkItem.java:100]: We also could compare `comment.author().id()` here, but I am not sure if it is a good idea to explicitly expose the id of mlbridge bot.

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java [hunk 2]
```diff
@@ -725,9 +725,20 @@ private String getStatusMessage(PullRequestCheckIssueVisitor visitor,
...
         progressBody.append(makeCollapsible("Using diff file", reviewUsingDiffsHelp()));
 
+        progressBody.append("\n\n### Webrev\n");
+        progressBody.append(getWebrevCommentLink());
         return progressBody.toString();
     }
 
+    private String getWebrevCommentLink() {
+        var webrevComment = comments.stream()
+                .filter(comment -> comment.author().username().equals("mlbridge[bot]"))
```
Comment (erikj79) [inline bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java:735]: We can't hard code the bot name here, it needs to be a configuration parameter for pr bot. See configuration of mlbridge for an example of how we configure user names to filter comments.

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckWorkItem.java (hunk 1 above)
Comment (erikj79) [reply bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckWorkItem.java:100]: The `commentString` looks like it's only containing comments from the currentUser (which is the bot user for the pr bot). I think we need to filter all comments again for the mlbridge user to find this comment.

**Review by erikj79 — COMMENTED** [review]
Overall a nice solution to this issue.

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckWorkItem.java (hunk 1 above)
Comment (zhaosongzs) [reply bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckWorkItem.java:100]: Yes, commentString only containing comments from the currentUser. So I actually filter all comments again in line 99.

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java (hunk 2 above)
Comment (zhaosongzs) [reply bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java:735]: Will fix it.

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckWorkItem.java (hunk 1 above)
Comment (erikj79) [reply bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckWorkItem.java]: Oh right, I misread that.

**Review by erikj79 — APPROVED** [review]

**Pull request closed without merging**